atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraClientPoolMetrics.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra.pool;
+
+import java.net.InetSocketAddress;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import com.codahale.metrics.Meter;
+import com.google.common.base.Supplier;
+import com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPool;
+import com.palantir.atlasdb.keyvalue.cassandra.CassandraClientPoolingContainer;
+import com.palantir.atlasdb.util.MetricsManager;
+
+public class CassandraClientPoolMetrics {
+    private final MetricsManager metricsManager = new MetricsManager();
+    private final RequestMetrics aggregateMetrics = new RequestMetrics(null);
+    private final Map<InetSocketAddress, RequestMetrics> metricsByHost = new HashMap<>();
+
+    public void deregisterMetrics() {
+        metricsManager.deregisterMetrics();
+    }
+
+    public void registerAggregateMetrics(Supplier<Integer> blacklistSize) {
+        metricsManager.registerMetric(
+                CassandraClientPool.class, "numBlacklistedHosts",
```

Comment:
May want to add a comment that we use `CassandraClientPool.class` here for backwards compatibility.